lib/ignore/ignore_test.go

```diff
@@ -1193,3 +1193,36 @@ func TestEmptyPatterns(t *testing.T) {
 		}
 	}
 }
+
+func TestWindowsLineEndings(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skip("Windows specific")
+	}
+
+	lines := "foo\nbar\nbaz\n"
+
+	ffs := fs.NewFilesystem(fs.FilesystemTypeBasic, ".")
+	m := New(ffs)
+	if err := m.Parse(strings.NewReader(lines), ".stignore"); err != nil {
+		t.Fatal(err)
+	}
+	if err := WriteIgnores(ffs, ".stignore", m.Lines()); err != nil {
```

Comment:
As far as I see this creates an `.stignore` in lib/ignores and doesn't remove it. Generally I'd also prefer `ioutil.TempDir`.